pkgs/tools/networking/dd-agent/datadog-agent.nix

```diff
@@ -41,8 +41,9 @@ in buildGoModule rec {
       "-X ${goPackagePath}/pkg/collector/py.pythonHome=${python}"
       "-r ${python}/lib"
     ];
+    buildTags = extraTags ++ lib.optionals withSystemd ["systemd"];
   in ''
-    buildFlagsArray=( "-tags" "ec2 systemd cpython process log secrets ${lib.concatStringsSep " " extraTags}" "-ldflags" "${ldFlags}")
+    buildFlagsArray=( "-tags" "ec2 cpython process log secrets ${lib.concatStringsSep " " buildTags}" "-ldflags" "${ldFlags}")
```

Comment:
This should use the nix options.

Reply:
I wasn't sure if this is the change you were after. If not, could you add some more details, or link to an example, please?

Reply:
It should use https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/go.section.md#tags-var-go-tags instead of bash variables.

Reply:
Right, thank you. Done both tags and ldflags since we're here.